FirebaseAdmin/FirebaseAdmin/Messaging/Util/AnalyticsLabelChecker.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace FirebaseAdmin.Messaging.Util
+{
+    /// <summary>
+    /// Checker for analytics label.
+    /// </summary>
+    public static class AnalyticsLabelChecker
+    {
+        private static ImmutableHashSet<char> alowedChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-~%".ToCharArray().ToImmutableHashSet();
+
+        /// <summary>
+        /// Checks anytytics labels an throw if not valid.
+        /// </summary>
+        /// <exception cref="ArgumentException">If analytics label does not match pattern.</exception>
+        /// <param name="analyticsLabel">Analytics label.</param>
+        public static void CheckAnalyticsLabelOrThrow(string analyticsLabel)
+        {
+            if (analyticsLabel.Length > 50)
```

Comment:
Can we just do a regex match like in Java?

```
Regex.IsMatch(analyticsLabel, pattern)
```

You should also handle the null case.

Reply:
`Regex.IsMatch(analyticsLabel, pattern)` works slower and uses more memory
![photo_2019-07-18_15-48-43](https://user-images.githubusercontent.com/13566100/61486533-1d7aa200-a9bd-11e9-91f0-8430bcbc7510.jpg)
here  benchmarks for different types of checked strings with different check ways

Reply:
I'd gladly accept a couple of milliseconds overhead if that means code is simpler and more readable. Your current implementation is O(N). Regex.IsMatch() is probably in the same order or at most O(N^2). But N is too small in this case to make a significant difference in any case. So I still prefer the Regex solution here.